### PR TITLE
demo-app: add occupancy to the demo app.

### PR DESCRIPTION
**Pull request by splindsay-92**
### Context

* [CHA-345]
*  The demo app did not use the occupancy capability of the chat SDK, so this PR adds it's usage.

### Description

* Added Occupancy component to render the occupancy metric on screen.  
* Added occupancy hook to receive occupancy events and update the metric.
* Added some basic styling.

Screenshot 2024-07-17 at 19 59 09

### Checklist

* [x] QA'd by the author.
* [ ] Unit tests created (if applicable).
* [ ] Integration tests created (if applicable).
* [x] Follow coding style guidelines found [here](https://github.com/ably/engineering/tree/main/best-practices).
* [ ] TypeDoc updated (if applicable).
* [ ] (Optional) Update documentation for new features.

### Testing Instructions (Optional)

* Spin up the demo app and open it in a few tabs, you should see the occupancy metric update accordingly.

[CHA-345]: https://ably.atlassian.net/browse/CHA-345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: demo/public/connected-persons-triangle-svgrepo-com.svg
Comment (AndyTWF) [inline demo/public/connected-persons-triangle-svgrepo-com.svg:1]: Just sanity checking - what was the source for this image, do we need to attribute etc?

File: demo/src/components/OccupancyComponent/index.ts
Comment (AndyTWF) [inline demo/src/components/OccupancyComponent/index.ts:1]: Empty file that we can get rid of?

File: demo/src/hooks/useOccupancy.ts
Comment (vladvelici) [inline demo/src/hooks/useOccupancy.ts:1]: Do we want to provide both metrics that we expose - connections and presenceMembers? Not sure if we want both numbers right now _"4324 online, 43 logged in"_

File: demo/public/connected-persons-triangle-svgrepo-com.svg
Comment (splindsay-92) [reply demo/public/connected-persons-triangle-svgrepo-com.svg:1]: https://www.svgrepo.com/svg/170177/connected-persons-triangle
Tis under the CCO license so we are all good without attributing :)

File: demo/src/components/OccupancyComponent/index.ts
Comment (splindsay-92) [reply demo/src/components/OccupancyComponent/index.ts:1]: Actually, I've added the export here so we keep consistency :)

File: demo/public/connected-persons-triangle-svgrepo-com.svg
Comment (AndyTWF) [reply demo/public/connected-persons-triangle-svgrepo-com.svg:1]: Perfect!

File: demo/src/hooks/useOccupancy.ts
Comment (splindsay-92) [reply demo/src/hooks/useOccupancy.ts:1]: I did think I'd leave this out and rely on the presence feature, but I'll add it in and make the distinction clear between connections and presence :)

**Review by AndyTWF — APPROVED** [review]

**Review by vladvelici — APPROVED** [review]

**Pull request merged**